### PR TITLE
Promote CSIDriver deployment to stable

### PR DESCRIPTION
**Pull request by Jiawei0227**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promote CSIDriver object deployment to GA

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Install CSIDriver object as part of driver deployment.
```

/cc @saad-ali 
/cc @verult 
/cc @saikat-royc

### CODE REVIEW

File: deploy/kubernetes/base/controller/cluster_setup.yaml [hunk 1]
```diff
@@ -272,3 +272,12 @@ roleRef:
   kind: Role
   name: csi-gce-pd-leaderelection-role
   apiGroup: rbac.authorization.k8s.io
+
```
Comment (saikat-royc) [inline deploy/kubernetes/base/controller/cluster_setup.yaml:275]: Lets keep this in a seperate file? given that cluster setup has only rbac rules? @msau42  thoughts

Comment (Jiawei0227) [reply deploy/kubernetes/base/controller/cluster_setup.yaml]: Done

**Comment by Jiawei0227** [comment]
Oops, sorry. Done

**Pull request merged**